pkg/handlers/post.go

```diff
@@ -1,16 +1,175 @@
 package handlers
 
 import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"log"
 	"net/http"
-
-	"github.com/Ubivius/microservice-template/pkg/data"
+	"net/url"
+	"strings"
 )
 
-// AddProduct creates a new product from the received JSON
-func (productHandler *ProductsHandler) AddProduct(responseWriter http.ResponseWriter, request *http.Request) {
-	productHandler.logger.Println("Handle POST Product")
-	product := request.Context().Value(KeyProduct{}).(*data.Product)
+// SignIn will fetch the acces token from Keycloak and return user data
+func (authHandler *AuthHandler) SignIn(responseWriter http.ResponseWriter, request *http.Request) {
+	signinPath := "http://localhost:8080/auth/realms/ubivius/protocol/openid-connect/token"
+
+	requestbody, _ := ioutil.ReadAll(request.Body)
+
+	username := ExtractValue(string(requestbody), "username")
+	password := ExtractValue(string(requestbody), "password")
+	log.Println(username)
+	log.Println(password)
+
+	data := url.Values{}
+	data.Set("client_id", "ubivius-client")
+	data.Set("grant_type", "password")
+	data.Set("client_secret", "7d109d2b-524f-4351-bfda-44ecad030eef")
+	data.Set("scope", "openid")
+	data.Set("username", username)
+	data.Set("password", password)
+
+	//SignIn
+	req, err := http.NewRequest("POST", signinPath, strings.NewReader(data.Encode()))
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+	access_token := ExtractValue(string(body), "access_token")
+	//Get user data
+	claims := ExtractClaims(access_token)
+	userId := claims["sub"]
+
+	userPath := "http://localhost:9091/users/" + userId.(string)
+
+	req, err = http.NewRequest("GET", userPath, nil)
+	if err != nil {
+		panic(err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	client = &http.Client{}
+	resp, err = client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, _ = ioutil.ReadAll(resp.Body)
+
+	jsonBody := map[string]string{}
+	err = json.Unmarshal(body, &jsonBody)
+	if err != nil {
+		panic(err)
+	}
+	jsonBody["accessToken"] = access_token
+	player, _ := json.Marshal(jsonBody)
+
+	log.Println(string(player))
+	responseWriter.WriteHeader(http.StatusOK)
+	_, err = responseWriter.Write(player)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// SignUp will register a new user in keycloak and in our user database
+func (authHandler *AuthHandler) SignUp(responseWriter http.ResponseWriter, request *http.Request) {
+	newUserPath := "http://localhost:8080/auth/admin/realms/ubivius/users"
+
+	requestbody, _ := ioutil.ReadAll(request.Body)
+	username := ExtractValue(string(requestbody), "username")
+	firstName := ExtractValue(string(requestbody), "firstName")
+	lastName := ExtractValue(string(requestbody), "lastName")
+	email := ExtractValue(string(requestbody), "email")
```

Comment:
Where is the password?

Reply:
j'ai ajouté la partie pour set le password